starter_code/elevator.js

```diff
@@ -1,19 +1,70 @@
 class Elevator {
-  constructor(){
-    this.floor      = 0;
-    this.MAXFLOOR   = 10;
-    this.requests   = [];
-  }
-
-  start() { }
-  stop() { }
-  update() { }
-  _passengersEnter() { }
-  _passengersLeave() { }
-  floorUp() { }
-  floorDown() { }
-  call() { }
-  log() { }
+  constructor() {
+    this.floor = 0;
+    this.MAXFLOOR = 10;
+    this.requests = []; // a list of pending requests (numbers)
+    this.direction = "";
+    this.waitingList = []; // people waiting for the elevator
+    this.passengers = []; // people currently in the elevator
+  }
+
+  start() {
+    this.startId = setInterval(() => {
+      this.update();
+    }, 1000);
+  }
+
+  stop() {
+    clearInterval(this.startId);
+  }
+
+  update() {
+    this._passengersEnter();
+    this._passengersLeave();
+    this.log();
+    this.floorUp();
```

Comment:
Of course it should be either `floorUp` or `floorDown`, depending on the current state of requests